portfolio/src/main/java/com/google/sps/servlets/CovidDataServlet.java

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.lang.NumberFormatException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns COVID-19 data as a JSON object, e.g. {"2020-01-27": [2820, 20]} */
+@WebServlet("/covid-data")
+public class CovidDataServlet extends HttpServlet {
+
+  private static class CovidData {
+    private final int totalCases;
+    private final int newCases;
```

Comment:
does this mean that since you're returning it direclty as the json, the variable names can be referred to in the js. basically if you change this variable name, then in the javascript the reference also changes right?

looking at your code: `covidTotals[date].totalCases, covidTotals[date].newCases]);`

Reply:
I believe so yes (like if I changed the variable names, then I would have to change it in the JS)

Reply:
a couple of things if we were working in google3:
-i guess this is the kind of use-case that having language-agnostic protos might be useful but i dont think externally there's a javascript- generarted code for protobufs either (but jspb and json aren't exactly equivalent).
- https://drive.google.com/file/d/0B-KFDiTrQe9lenFxcXVFbURDeDA/view we could also set-up linters so that the names remain in sync with each other.

just fyi's, no need to use any of them for this PR but good to read-up on.